### ISSUE TITLE
HashNet on CUB200

### ISSUE THREAD

**Issue by zhouxiaohang** [thuml/HashNet#38]
Hi, I've tried to use HashNet to Fine-grained recognition, so I adopted the PyTorch code to CUB200 dataset with finetuned ResNet50, but I can't make the loss converge. I've tried different optimizers like SGD, Adam, RMSprop, and different class_num values like 1.0 and 200.0, different lr values from 1e-5 to 1e-3.

Here is a set of parameters which I tried:

```
python train.py \
    --dataset cub200 \
    --prefix resnet50_hashnet \
    --hash_bit 64 \
    --net ResNet50 \
    --lr 1e-5 \
    --class_num 1.0

{'l_weight': 1.0, 'q_weight': 0, 'l_threshold': 15.0, 'sigmoid_param': 0.15625, 'class_num': 1.0}{'type': 'RMSprop', 'optim_params': {'lr': 1.0, 'weight_decay': 1e-05}, 'lr_type': 'step', 'lr_param': {'init_lr': 1e-05, 'gamma': 0.5, 'step': 2000}}

```

But the training loss is always around 0.69, and mAP is extremely as low as 0.04.

No matter what parameters are used, mAP is always lower than 0.05. Intuitively, this's not reasonable.

Have you ever apply HashNet on CUB200? Do you have any ideas on it? Thanks.

**Comment by 633NuLL** [comment]
Hi,did you solve the problem yet?I also met same problem on my dataset.
